### PR TITLE
0.16.x

### PR DESCRIPTION
**Pull request by christyjacob4**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: app/views/docs/permissions.phtml [hunk 1]
```diff
@@ -23,18 +23,17 @@
 <div class="notice calm margin-top-large margin-bottom-large">
     <h4>Note</h4>
     
-    <p>You will need an active session to handle database transactions for your collections.</p>
-    <p>If you have rules where you allow for wild cards or role:guest in a collection, you can create an Anonymous Session before you make the API call.</p>
+    <p>You will need an active session to perform create, update, and delete type actions in your collections. Even if create, update, and delete permissions are granted to "Role.any()", you still need to <a href="/docs/client/account#accountCreateAnonymousSession" target="_blank">create an anonymous session</a> before you can perform these actions.</p>
```
Comment (abnegate) [inline app/views/docs/permissions.phtml:26]: > You will need an active session to perform create, update, and delete type actions in your collections

This is no longer true

File: app/views/docs/permissions.phtml [hunk 2]
```diff
@@ -23,18 +23,17 @@
...
 </div>
 
 <h2 id="server-integration"><a href="/docs/permissions#server-integration">Server Integration</a></h2>
 
-<p>A server or <a href="/docs/admin" target="_blank">admin integration</a> can be used for increased flexibility. When using a server integration in combination with the proper <a href="/docs/keys#scopes" target="_blank">API scopes</a>, you can have both read and write access to any of your project resources regardless of their permissions.</p>
+<p>A server or <a href="/docs/admin">admin integration</a> can be used for increased flexibility. When using a Server SDK in combination with the proper <a href="/docs/keys#scopes">API key scopes</a>, you can have all types of access to any of your project resources regardless of their permissions.</p>
```
Comment (abnegate) [inline app/views/docs/permissions.phtml:31]:
```suggestion
<p>A server or <a href="/docs/admin">admin integration</a> can be used for increased flexibility. When using a Server SDK in combination with the proper <a href="/docs/keys#scopes">API key scopes</a>, you can have any type of access to any of your project resources regardless of their permissions.</p>
```

File: app/views/docs/permissions.phtml [hunk 3]
```diff
@@ -44,60 +43,62 @@
...
+        <td>Grants access to any member of the specific team. To gain access to this permission, the user must be the team creator (owner), or receive and accept an invitation to join this team.</td>
+    </tr>
+    <tr>
+        <td>Role.team([TEAM_ID], [ROLE])</td>
+        <td>Grants access to any member who possesses a specific role in a team. To gain access to this permission, the user must be a member of the specific team and have the given role assigned to them. Team roles can be assigned when inviting a user to become a team member.</td>
+    </tr>
+    </tbody>
+</table>
+
+<h2 id="permission-types"><a href="/docs/permissions#permission-types">Permission Types</a></h2>
```
Comment (abnegate) [inline app/views/docs/permissions.phtml:72]: I think this block should come before `Permission Roles`

File: app/views/docs/permissions.phtml [hunk 4]
```diff
@@ -44,60 +43,62 @@
...
+        <td style="width: 180px">Type</td>
+        <td>Description</td>
     </tr>
+    </thead>
+    <tbody>
     <tr>
-        <td>team:[TEAM_ID]/[ROLE]</td>
-        <td>Access to any member who possesses a specific role in a team. To gain access to this permission, the user must be a member of the specific team and have the given role assigned to him or her. Team roles can be assigned when inviting a user to become a team member.</td>
+        <td>Permission.read()</td>
+        <td>Access to view a resource. Does not apply to functions.</td>
```
Comment (abnegate) [inline app/views/docs/permissions.phtml:84]: > Does not apply to functions.

I think we can remove all instances of this as functions won't use permissions, only roles

Comment (abnegate) [inline app/views/docs/permissions.phtml:84]:
```suggestion
        <td>Access to read a resource.</td>
```

File: app/views/docs/permissions.phtml [hunk 5]
```diff
@@ -44,60 +43,62 @@
...
+        <td>Access to remove a resource. Does not apply to functions.</td>
     </tr>
     <tr>
-        <td>
-            *
-            <p class="margin-top-small text-fade text-danger">version <= 0.11 (deprecated)</p>
-        </td>
-        <td>Wildcard permission. Gives anyone read or write access.</td>
+        <td>Permission.write()</td>
+        <td>Alias to grant all of create, update, and delete access.</td>
```
Comment (abnegate) [inline app/views/docs/permissions.phtml:100]: > create, update, and delete access.

Will only grant update, delete for documents and files

File: app/views/docs/permissions.phtml [hunk 6]
```diff
@@ -109,10 +110,10 @@
 
 <h3 id="example-1"><a href="/docs/permissions#example-1">Example #1 - Basic Usage</a></h3>
 
-<p>In the following example, we are creating a document that can be read by everyone and only be edited, or deleted by a user with a UID <span class="tag">user:5c1f88b42259e</span>.</p>
+<p>In the following example, we are creating a document that can be viewed by anyone, edited by writers or admins, and deleted by a user with the user ID <span class="tag">user:5c1f88b42259e</span> or administrators.</p>
```
Comment (abnegate) [inline app/views/docs/permissions.phtml:113]:
```suggestion
<p>In the following example, we are creating a document that can be read by anyone, edited by writers or admins, and deleted by administrators or a user with the user ID <span class="tag">user:5c1f88b42259e</span>.</p>
```

File: app/views/docs/permissions.phtml [hunk 7]
```diff
@@ -123,27 +124,28 @@ client
...
-});
-
-</code></pre>
+});</code></pre>
 </div>
 
 <h3 id="example-2"><a href="/docs/permissions#example-2">Example #2 - Team Roles</a></h3>
 
-<p>In the following example, we are creating a document that can be read-only by members of <span class="tag">team:5c1f88b87435e</span> and can only be edited, or deleted by members of the same team that possesses the role <span class="tag">owner</span>.</p>
+<p>In the following example, we are creating a document that can be read-only by members of the team with ID <span class="tag">5c1f88b87435e</span> and can only be edited or deleted by members of the same team that possesses the team role <span class="tag">owner</span>.</p>
```
Comment (abnegate) [inline app/views/docs/permissions.phtml:145]:
```suggestion
<p>In the following example, we are creating a document that can be read by members of the team with ID <span class="tag">5c1f88b87435e</span> and can only be edited or deleted by members of the same team that possess the team role <span class="tag">owner</span>.</p>
```

File: app/views/docs/permissions.phtml [hunk 8]
```diff
@@ -123,27 +124,28 @@ client
...
 let promise = databases.createDocument(
-    '[COLLECTION_ID]',
-    {'actorName': 'Chris Evans', 'height': 183},
-    ['role:all'], // Anyone can view this document
-    ['user:5c1f88b42259e'] // You can only grant permissions you own, this must be the current user UID
+    '[COLLECTION_ID]', // Collection ID
+    {'actorName': 'Chris Evans', 'height': 183}, // Data
+    [ // Permissions
+        Permission.read(Role.any()), // Anyone can view this document
+        Permission.edit([Role.team("writers"), Role.team("admin")]), // Only writers and admins can edit this document
```
Comment (abnegate) [inline app/views/docs/permissions.phtml:131]: We now only allow one role per permission

```suggestion
        Permission.update(Role.team("writers")), // Writers can update this document
        Permissions.update(Role.team("admin")), // Admins can update this document
```

File: app/views/docs/permissions.phtml [hunk 9]
```diff
@@ -123,27 +124,28 @@ client
...
+        Permission.delete([Role.user("5c1f88b42259e"), Role.team("admin")]) // Only user 5c1f88b42259e and admins can delete this document
```
Comment (abnegate) [inline app/views/docs/permissions.phtml:132]:
```suggestion
       Permission.delete(Role.user("5c1f88b42259e")) // User 5c1f88b42259e can delete this document
       Permission.delete(Role.team("admin")) // Admins can delete this document
```

File: app/views/docs/permissions.phtml [hunk 10]
```diff
@@ -156,68 +158,16 @@ const databases = new Databases(client, '[DATABASE_ID]');
 let promise = databases.createDocument(
     '[COLLECTION_ID]',
     {'actorName': 'Chris Evans', 'height': 183},
-    ['team:5c1f88b87435e'], // The user must be a team member to grant this permission
-    ['team:5c1f88b87435e/owner']); // The user must be a team owner to grant this permission
+    [
+        Permission.read(Role.team("5c1f88b87435e")), // Only users of team 5c1f88b87435e can read the document
+        Permission.edit(Role.team("5c1f88b87435e", "owner")), // Only users of team 5c1f88b87435e with the role owner can edit the document
```
Comment (abnegate) [inline app/views/docs/permissions.phtml:163]:
```suggestion
        Permission.update(Role.team("5c1f88b87435e", "owner")), // Only users of team 5c1f88b87435e with the role owner can update the document
```

File: app/views/docs/realtime.phtml [hunk 11]
```diff
@@ -118,8 +118,8 @@ let subscription = realtime.subscribe(channels: ["files"]) { message in
 
 <div class="notice margin-top-large margin-bottom-large">
     <h3>Permissions</h3>
-    <p>All subscriptions are secured by the <a href="/docs/permissions">permissions</a> offered by Appwrite, meaning a user will only receive updates to resources to which he has read permission.</p>
-    <p>Using <span class="tag">role:all</span> on read permissions will allow any client to receive updates.</p>
+    <p>All subscriptions are secured by the <a href="/docs/permissions">permissions system</a> offered by Appwrite, meaning a user will only receive updates to resources to which they has permission to read.</p>
```
Comment (abnegate) [inline app/views/docs/realtime.phtml:121]:
```suggestion
    <p>All subscriptions are secured by the <a href="/docs/permissions">permissions system</a> offered by Appwrite, meaning a user will only receive updates to resources they have permission to access.</p>
```

File: app/views/docs/service.phtml [hunk 12]
```diff
@@ -314,9 +314,9 @@ $paramSDK = (!empty($sdk)) ? '?sdk='.$sdk : '';
                             <h3 class="margin-bottom margin-top-small">Authentication</h3>
 
                             <?php if(array_key_exists('Key', $security) && array_key_exists('JWT', $security)): ?>
-                                <p>To access this route, init your SDK with your project unique ID and <a href="/docs/keys" target="_blank">API Key</a> secret token. Make sure your API Key is granted with access to the <u>"<?php echo $scope; ?>"</u> permission scope. You can also authenticate using a valid <a href="/docs/client/account#accountCreateJWT" target="_blank">JWT</a> and perform actions on behalf of your user.</p>
+                                <p>To access this route, init your SDK with your project unique ID and <a href="/docs/keys">API Key</a> secret token. Make sure your API Key is granted with access to the <u>"<?php echo $scope; ?>"</u> permission scope. You can also authenticate using a valid <a href="/docs/client/account#accountCreateJWT">JWT</a> and perform actions on behalf of your user.</p>
```
Comment (abnegate) [inline app/views/docs/service.phtml:317]:
```suggestion
                                <p>To access this route, init your SDK with your project unique ID and an <a href="/docs/keys">API Key</a>. Make sure your API Key is created with the <u>"<?php echo $scope; ?>"</u> scope. You can also authenticate using a valid <a href="/docs/client/account#accountCreateJWT">JWT</a> and perform actions on behalf of your user.</p>
```

File: app/views/docs/service.phtml [hunk 13]
```diff
@@ -314,9 +314,9 @@ $paramSDK = (!empty($sdk)) ? '?sdk='.$sdk : '';
...
                             <?php elseif(array_key_exists('Key', $security)): ?>
-                                <p>To access this route, init your SDK with your project unique ID and <a href="/docs/keys" target="_blank">API Key</a> secret token. Make sure your API Key is granted with access to the <u>"<?php echo $scope; ?>"</u> permission scope.</p>
+                                <p>To access this route, init your SDK with your project unique ID and <a href="/docs/keys">API Key</a> secret token. Make sure your API Key is granted with access to the <u>"<?php echo $scope; ?>"</u> permission scope.</p>
```
Comment (abnegate) [inline app/views/docs/service.phtml:319]:
```suggestion
                                <p>To access this route, init your SDK with your project unique ID and an <a href="/docs/keys">API Key</a>. Make sure your API Key is create with the <u>"<?php echo $scope; ?>"</u> scope.</p>
```

File: app/views/docs/storage.phtml [hunk 14]
```diff
@@ -13,21 +13,21 @@
...
-
-<h4>File-Level Permissions</h4>
-<p>With this permission model, you have granular access control over every file, and users will only be able to access files for which they have explicit permissions. This permission model requires file permissions, and bucket permissions are optional.
-</p>
-
-<h4>Bucket-Level Permissions</h4>
-<p>With bucket-level permissions, you assign permissions once for every file in the buckets' settings - users with "read" access to the bucket can see all files in that bucket. This permission model requires you to set bucket permissions on the bucket settings page. File permissions are optional and will not be evaluated even when set.</p>
+<p>Storage Service allows you to configure permissions at both the bucket level and the file level. When a user has the appropriate type of <a href="/docs/permissions/">access permissions</a> granted at **either** the bucket or the file level, they will be able to access the file. If the permission field is left empty, no one can access the file.</p>
```
Comment (abnegate) [inline app/views/docs/storage.phtml:16]:
```suggestion
<p>The Storage Service allows you to configure permissions at both the bucket level and the file level. When a user has the appropriate type of <a href="/docs/permissions/">access permissions</a> granted at **either** the bucket or the file level, they will be able to access the file. If the permission field is left empty, no one can access the file.</p>
```

File: app/views/docs/storage.phtml [hunk 15]
```diff
@@ -13,21 +13,21 @@
...
 
 <div class="notice margin-top-large margin-bottom-large">
     <h5>Active Session Required</h5>
 
-    <p>For security purposes, an active Account session is required to create resources. So, even if wildcard write permissions are set, only logged-in users can create files in a bucket. If you require this behavior for your app, create an <a href="/docs/client/account?sdk=web-default#accountCreateAnonymousSession" target="_blank">anonymous session</a> first. An active session helps Appwrite enforce rate-control limits and protect your projects from intensive write operations.</p>
+    <p>For security purposes, an active Account session is required to create resources. So, even if wildcard write permissions are set, only logged-in users can create files in a bucket. If you require this behavior for your app, create an <a href="/docs/client/account?sdk=web-default#accountCreateAnonymousSession">anonymous session</a> first. An active session helps Appwrite enforce rate-control limits and protect your projects from intensive write operations.</p>
```
Comment (abnegate) [inline app/views/docs/storage.phtml:21]: > For security purposes, an active Account session is required to create resources.

This is no longer true

File: app/views/docs/storage.phtml [hunk 16]
```diff
@@ -13,21 +13,21 @@
...
+<h5>File Level Permissions</h5>
+<p>File level permissions grant access to individual files. File level permissions are only enabled if File Security is enabled in the settings of your bucket.
+</p>
+
+<h5>Bucket Level Permissions</h5>
+<p>Bucket level permissions apply to every file in the bucket.</p>
+
 <h4><a href="/docs/storage#bucketConfig" id="bucketConfig">More Bucket Configurations</a></h4>
 
 <p>Unlike collections, storage buckets have more configuration options regarding the type and size of the files in the bucket, encryption, and anti-virus scanning. If you look at the bucket settings or the REST API example above, you can find these configurations. Let's look at what those are:</p>
```
Comment (abnegate) [inline app/views/docs/storage.phtml:33]: No need for a comparison to collections here

```suggestion
<p>Storage buckets have many configuration options, including the type and maximum size of files in the bucket, whether encryption or anti-virus is enabled,  and the compression algorithm to use. If you look at the bucket settings or the REST API example above, you can find these configurations. Let's look at what those are:</p>
```

Comment (abnegate) [inline app/views/docs/storage.phtml:33]: Looks like we might need to add some updates below here detailing the new `compression` attribute on buckets

**Review by abnegate — CHANGES_REQUESTED** [review]

File: app/views/docs/databases.phtml [hunk 17]
```diff
@@ -33,25 +33,19 @@
...
 </p>
 
 <div class="notice margin-top-large margin-bottom-large">
     <h5>Active Session Required</h5>
 
-    <p>For security purposes, an active Account session is required to create resources. So, even if wildcard write permissions are set, only logged-in users can create documents in a collection. If you require this behavior for your app, create an <a href="/docs/client/account?sdk=web-default#accountCreateAnonymousSession" target="_blank">anonymous session</a> first. An active session helps Appwrite enforce rate-control limits and protect your projects from intensive write operations.</p>
+    <p>For security purposes, an active Account session is required to create resources. So, even if wildcard write permissions are set, only logged-in users can create documents in a collection. Create an <a href="/docs/client/account?sdk=web-default#accountCreateAnonymousSession">anonymous session</a> if you require this behavior for your app. An active session helps Appwrite enforce rate-control limits and protect your projects from intensive write operations.</p>
```
[the remark this answers is not in the document]
Comment (christyjacob4) [reply app/views/docs/databases.phtml:42]: addressed

**Pull request merged**